lisa/testselector.py

```diff
@@ -125,6 +125,7 @@ def _apply_settings(
     applied_case_data.use_new_environment = (
         applied_case_data.use_new_environment or case_runbook.use_new_environment
     )
+    applied_case_data.timeout = case_runbook.timeout or applied_case_data.timeout
```

Comment:
When is this change needed? If it needs to extend the timeout, it should take max of both. If it wants to run test cases faster, it should take min of them. They are conflict goals.

For the `or` condition, it takes case_runbook higher priority, for some designed long running test cases, they will mostly failed. It's not right to overwrite case settings always.